src/test/java/AcmeShop.java

```diff
@@ -88,16 +80,17 @@ public void addingToCart() {
         WebElement updateButton = webdriver.findElement(By.name("update_cart_item"));
         updateButton.click();
 
+        Thread.sleep(2000);
+
             //Проверяем, отобразилось ли в корзине правильное количество(3)
         WebElement cartQuantity = webdriver.findElement(By.cssSelector("input[type=\"number\"]"));
-        String quantityFact = cartQuantity.getAttribute("value"); //иногда не проходит в этом месте, так как тоже не успевает отобразиться и считать
+        String quantityFact = cartQuantity.getAttribute("value");
         Assert.assertEquals(quantityFact, "3");
 
             //Проверяем, отобразилось ли правильное количество(3) в таблице ниже
         WebElement gridQuantity = webdriver.findElement(By.cssSelector("td[style=\"text-align: center;\"]"));
         String quantityGrid = gridQuantity.getText();
-        //Assert.assertEquals(quantityGrid, "3"); // Здесь тест провалится, так как тоже нужно добавить ожидание.
-                                                        // Так как проходит какое-то время, пока update долетит до таблицы
+        Assert.assertEquals(quantityGrid, "3");
 
             // Для тренировки DragAndDrop перетянем изображение уточки в поле комментария (появится url в нем)
```

Comment:
tricky task in general. Here's the comment on how it should work:
moveToElement moves the cursor to the center of the element, so the offset you're setting is performed from the element center, so setting width/height as offset will move the cursor out of the element boundaries.
What you should do instead is set width/2 and height/2.
Also please note that this will get you to the very corner of the element which is not interactive (you can check it manually, if you place the cursor over the very bottom right corner it won't drag. So what you need to do is move it slightly up and left to hit that sweet spot that's draggable so your final offset should be:
(width/2)-5 and (height/2)-5
Lastly: you can't drag outside of the current window with Selenium, that's why it's throwing an exception when you're trying to drag to 5 heights so you can only slightly drag it down. But from testing point of view this should be ok, try dragging to 10 and 10 pixels and then assert that the size has properly changed

Reply:
It does not work in this edition either. I tried minus 1, 2 5, etc